lib/jxl/dec_reconstruct.cc

```diff
@@ -287,6 +287,114 @@ void FloatToRGBA8(const Image3F& input, const Rect& input_rect, bool is_rgba,
   }
 }
 
+void DoYCbCrUpsampling(size_t hs, size_t vs, ImageF* plane_in, const Rect& rect,
+                       const Rect& frame_rect, const FrameDimensions& frame_dim,
+                       ImageF* plane_out, const LoopFilter& lf, ImageF* temp) {
+  // The pixel in (xoff, yoff) is the origin of the downsampling coordinate
+  // system.
+  size_t xoff = PassesDecoderState::kGroupDataXBorder;
+  size_t yoff = PassesDecoderState::kGroupDataYBorder;
+  // This may over-copy, but it should always be safe to do so.
+  size_t y0 = rect.y0() - lf.Padding();
+  size_t y1 = rect.y0() + rect.ysize() + lf.Padding();
+  size_t x0 = rect.x0() - lf.Padding();
```

Comment:
Is it possible for lf.Padding() to be > rect.xo()?

Reply:
Actually I mean whether it can be >= rect.x0() given the x0 - 1 below

Reply:
it shouldn't be, IIRC `rect.x0() >= RoundUpTo(lf.Padding(), 8) + 8`